### ISSUE TITLE
broadcasting over udp causes warning

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10933]
|<img alt="antony's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| antony reported|
|-|-|
|Trac ID|trac#1363|
|Type|enhancement|
|Created|2005-12-15 00:54:28Z|
|Branch|https://github.com/twisted/twisted/tree/udp-broadcast-warning-1363|

Attachments:

* [udp.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/c18/c18557ded94f94dc63cbcec520f2feeb823c4e3d/d1ce7d332b1c0022e31b0fb28a880d25e35ec168.diff) (515 bytes) - added by antony on 2005-12-15 00:54:27Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__1363 1363
type__enhancement enhancement
reporter__antony antony
priority__normal normal
milestone__ 
branch__branches_udp_broadcast_warning_1363 branches/udp-broadcast-warning-1363
branch_author__jesstess jesstess
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1134608068000000 1134608068000000
changetime__1267380533000000 1267380533000000
version__ 
owner__ 
cc__itamarst cc__antony cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="antony's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|antony commented|
|-|-|

```
#!html
<pre>
when sending messages over udp to host "&lt;broadcast>", Twisted complains it is
not an IP address and that it should be resolved first. This is obviously
unnecessary. The attached patch fixes it.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
Uh. What is address "&lt;broadcast>"? Where is it documented?
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="antony's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|antony commented|
|-|-|

```
#!html
<pre>
Well, I don't know much about it myself, and it seems to be extremely poorly
documented. Here are some references:

1. Here '255.255.255.255' is used:
http://www.knowdotnet.com/articles/udpbroadcastmessage.html

2. This is a module in Pyro:
http://cvs.sourceforge.net/viewcvs.py/pyro/Pyro/Pyro/naming.py?view=markup
And here's a quote from there: """
# No host specified. Use broadcast mechanism
	if os.name=='java':
		# Jython handling for broadcast addrs
		destination1 = ('255.255.255.255', port1)
		destination2 = ('255.255.255.255', port2)
	else:
		# regular python broadcast socket
		destination1 = ('&lt;broadcast>', port1)
		destination2 = ('&lt;broadcast>', port2)
"""

3. Try """socket.gethostbyname("&lt;broadcast>")""". It returns '255.255.255.255'.
From looking at the Python source code, the conversion doesn't seem to be blocking.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
So, can't you just use 255.255.255.255 in your code?
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="antony's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|antony commented|
|-|-|

```
#!html
<pre>
I can, but why prevent the seemingly legitimate alternative of "&lt;broadcast>"?
The Python sources indicate that the two possibilities are simply the same, so
if one is accepted, the other shouldn't cause a warning saying that it should be
resolved first, which is quite irrelevant.

Also, since this thing is poorly documented, if some poor user finds code using
"&lt;broadcast>" and wants to adapt it to Twisted, they're going to run into
needless trouble trying to address the issue of this warning, get their time
wasted, and possibily entertain a fleeting emotion of resentment towards
Twisted, all very negative outcomes! :)
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

This would be good.  The patch needs a unit test, though.  It should be pretty straightforward.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_8"></a>|@jesstess commented|
|-|-|

(In [28551]) Branching to 'udp-broadcast-warning-1363'

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_9"></a>|@jesstess commented|
|-|-|

(In [28552]) Add unit test for no deprecation warning in <broadcast> case.

refs [#10933](https://github.com/twisted/twisted/issues/10933)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_10"></a>|@jesstess removed owner|
|-|-|

udp.py has some coding standards issues. Outside of write() I only fixed the locations of triple-quotes in docstrings since that happens to particularly bother me.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_11"></a>|@therve set owner to @jesstess|
|-|-|

Looks good. Please make sure it runs correctly with a buildbot run, then merge.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_12"></a>|@jesstess set status to `closed`|
|-|-|

(In [28577]) Merge udp-broadcast-warning-1363

Authors: antony, jesstess
Reviewers: exarkun, therve
Fixes: [#10933](https://github.com/twisted/twisted/issues/10933)

'<broadcast>' is an alternative way to say '255.255.255.255'
(socket.gethostbyname("<broadcast>") returns '255.255.255.255'), so
don't emit a deprecation warning in twisted.internet.udp.Port.write
about passing hostnames instead of IP addresses for this special
'<broadcast>' case.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
